In [1]:
import numpy as np
import pandas as pd
import re
import requests
from bs4 import BeautifulSoup

import matplotlib.pyplot as plt
import seaborn as sns

import time

from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = 'all'

In [75]:
df = pd.read_csv(r"C:\Users\devender bahadur\Downloads\Web Scrabing Final Project\main_df.csv")

In [76]:
df

Unnamed: 0                       Title Year of Release Duration in min  \
0             0       \n1.\nThe Godfather\n         1972)\n         175 min   
1             1          \n2.\nGoodfellas\n         1990)\n         145 min   
2             2        \n3.\nPulp Fiction\n         1994)\n         154 min   
3             3  \n4.\nThe Usual Suspects\n         1995)\n         106 min   
4             4      \n5.\nApocalypse Now\n         1979)\n         147 min   
..          ...                         ...             ...             ...   
995         995          \n996.\nBlade II\n         2002)\n         117 min   
996         996      \n997.\nFalling Down\n         1993)\n         113 min   
997         997    \n998.\nEyes Wide Shut\n         1999)\n         159 min   
998         998     \n999.\nThe Lost Boys\n         1987)\n          97 min   
999         999      \n1,000.\nDeadpool 2\n         2018)\n         119 min   

                                       Genre  Rating Votes in Views  \
0                 \nCrime, Drama                 9.2        231,110   
1      \nBiography, Crime, Drama                 8.7        382,167   
2                 \nCrime, Drama                 8.9         84,108   
3        \nCrime, Drama, Mystery                 8.5         27,683   
4          \nDrama, Mystery, War                 8.5        413,088   
..                                       ...     ...            ...   
995     \nAction, Horror, Sci-Fi                 6.7        212,881   
996       \nAction, Crime, Drama                 7.6        185,112   
997   \nDrama, Mystery, Thriller                 7.5        328,196   
998             \nComedy, Horror                 7.2        137,754   
999  \nAction, Adventure, Comedy                 7.7        547,316   

    Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  Sci-Fi  \
0             $14.67M       0       0      1      1       0         0       0   
1            $353.01M       0       0      1      1       0         0       0   
2             $10.00M       0       0      1      1       0         0       0   
3              $0.05M       0       0      1      1       0         0       0   
4             $46.89M       0       0      1      0       0         0       0   
..                ...     ...     ...    ...    ...     ...       ...     ...   
995           $82.35M       1       1      0      0       0         0       1   
996           $40.90M       1       0      1      1       0         0       0   
997           $55.69M       0       0      1      0       0         1       0   
998           $32.22M       0       1      0      0       1         0       0   
999          $324.59M       1       0      0      0       1         0       0   

     Mystery  Adventure  Biography  war  
0          0          0          0    0  
1          0          0          1    0  
2          0          0          0    0  
3          1          0          0    0  
4          1          0          0    1  
..       ...        ...        ...  ...  
995        0          0          0    0  
996        0          0          0    0  
997        1          0          0    0  
998        0          0          0    0  
999        0          1          0    0  

[1000 rows x 19 columns]

# We are removing the Unnecessary items in the data frame

# Data cleaning

In [77]:
df['Genre'] = df['Genre'].apply(lambda x:x.replace('\n',''))

In [78]:
df['Duration in min'] = df['Duration in min'].apply(lambda x:x.replace('min',''))

In [79]:
df['Year of Release'] = df['Year of Release'].apply(lambda x:x.replace(')\n',''))

In [80]:
df['Title'] = df['Title'].apply(lambda x:x.replace('\n',''))

In [81]:
df['Votes in Views'] = df['Votes in Views'].apply(lambda x:x.replace(',',''))

In [82]:
df

Unnamed: 0                 Title Year of Release Duration in min  \
0             0       1.The Godfather            1972            175    
1             1          2.Goodfellas            1990            145    
2             2        3.Pulp Fiction            1994            154    
3             3  4.The Usual Suspects            1995            106    
4             4      5.Apocalypse Now            1979            147    
..          ...                   ...             ...             ...   
995         995          996.Blade II            2002            117    
996         996      997.Falling Down            1993            113    
997         997    998.Eyes Wide Shut            1999            159    
998         998     999.The Lost Boys            1987             97    
999         999      1,000.Deadpool 2            2018            119    

                                     Genre  Rating Votes in Views  \
0                 Crime, Drama                 9.2         231110   
1      Biography, Crime, Drama                 8.7         382167   
2                 Crime, Drama                 8.9          84108   
3        Crime, Drama, Mystery                 8.5          27683   
4          Drama, Mystery, War                 8.5         413088   
..                                     ...     ...            ...   
995     Action, Horror, Sci-Fi                 6.7         212881   
996       Action, Crime, Drama                 7.6         185112   
997   Drama, Mystery, Thriller                 7.5         328196   
998             Comedy, Horror                 7.2         137754   
999  Action, Adventure, Comedy                 7.7         547316   

    Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  Sci-Fi  \
0             $14.67M       0       0      1      1       0         0       0   
1            $353.01M       0       0      1      1       0         0       0   
2             $10.00M       0       0      1      1       0         0       0   
3              $0.05M       0       0      1      1       0         0       0   
4             $46.89M       0       0      1      0       0         0       0   
..                ...     ...     ...    ...    ...     ...       ...     ...   
995           $82.35M       1       1      0      0       0         0       1   
996           $40.90M       1       0      1      1       0         0       0   
997           $55.69M       0       0      1      0       0         1       0   
998           $32.22M       0       1      0      0       1         0       0   
999          $324.59M       1       0      0      0       1         0       0   

     Mystery  Adventure  Biography  war  
0          0          0          0    0  
1          0          0          1    0  
2          0          0          0    0  
3          1          0          0    0  
4          1          0          0    1  
..       ...        ...        ...  ...  
995        0          0          0    0  
996        0          0          0    0  
997        1          0          0    0  
998        0          0          0    0  
999        0          1          0    0  

[1000 rows x 19 columns]

# Checking that Nan values are present or not.

In [83]:
df.isna().sum()

Unnamed: 0            0
Title                 0
Year of Release       0
Duration in min       0
Genre                 0
Rating                0
Votes in Views        0
Gross in Millions    60
Action                0
Horror                0
Drama                 0
Crime                 0
Comedy                0
Thriller              0
Sci-Fi                0
Mystery               0
Adventure             0
Biography             0
war                   0
dtype: int64

#### There are 60 nan values in the gross in millions columns

# Checking the d-types of each column

In [84]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 19 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         1000 non-null   int64  
 1   Title              1000 non-null   object 
 2   Year of Release    1000 non-null   object 
 3   Duration in min    1000 non-null   object 
 4   Genre              1000 non-null   object 
 5   Rating             1000 non-null   float64
 6   Votes in Views     1000 non-null   object 
 7   Gross in Millions  940 non-null    object 
 8   Action             1000 non-null   int64  
 9   Horror             1000 non-null   int64  
 10  Drama              1000 non-null   int64  
 11  Crime              1000 non-null   int64  
 12  Comedy             1000 non-null   int64  
 13  Thriller           1000 non-null   int64  
 14  Sci-Fi             1000 non-null   int64  
 15  Mystery            1000 non-null   int64  
 16  Adventure          1000 n

In [85]:
df.describe()

Unnamed: 0       Rating      Action       Horror       Drama  \
count  1000.000000  1000.000000  1000.00000  1000.000000  1000.00000   
mean    499.500000     7.552200     0.25600     0.092000     0.57300   
std     288.819436     0.560758     0.43664     0.289171     0.49489   
min       0.000000     5.300000     0.00000     0.000000     0.00000   
25%     249.750000     7.200000     0.00000     0.000000     0.00000   
50%     499.500000     7.600000     0.00000     0.000000     1.00000   
75%     749.250000     7.900000     1.00000     0.000000     1.00000   
max     999.000000     9.300000     1.00000     1.000000     1.00000   

             Crime       Comedy    Thriller       Sci-Fi      Mystery  \
count  1000.000000  1000.000000  1000.00000  1000.000000  1000.000000   
mean      0.290000     0.282000     0.19300     0.089000     0.113000   
std       0.453989     0.450198     0.39485     0.284886     0.316751   
min       0.000000     0.000000     0.00000     0.000000     0.000000   
25%       0.000000     0.000000     0.00000     0.000000     0.000000   
50%       0.000000     0.000000     0.00000     0.000000     0.000000   
75%       1.000000     1.000000     0.00000     0.000000     0.000000   
max       1.000000     1.000000     1.00000     1.000000     1.000000   

         Adventure    Biography          war  
count  1000.000000  1000.000000  1000.000000  
mean      0.181000     0.105000     0.024000  
std       0.385211     0.306707     0.153126  
min       0.000000     0.000000     0.000000  
25%       0.000000     0.000000     0.000000  
50%       0.000000     0.000000     0.000000  
75%       0.000000     0.000000     0.000000  
max       1.000000     1.000000     1.000000

In [86]:
df.dropna(inplace=True)

In [87]:
df

Unnamed: 0                 Title Year of Release Duration in min  \
0             0       1.The Godfather            1972            175    
1             1          2.Goodfellas            1990            145    
2             2        3.Pulp Fiction            1994            154    
3             3  4.The Usual Suspects            1995            106    
4             4      5.Apocalypse Now            1979            147    
..          ...                   ...             ...             ...   
995         995          996.Blade II            2002            117    
996         996      997.Falling Down            1993            113    
997         997    998.Eyes Wide Shut            1999            159    
998         998     999.The Lost Boys            1987             97    
999         999      1,000.Deadpool 2            2018            119    

                                     Genre  Rating Votes in Views  \
0                 Crime, Drama                 9.2         231110   
1      Biography, Crime, Drama                 8.7         382167   
2                 Crime, Drama                 8.9          84108   
3        Crime, Drama, Mystery                 8.5          27683   
4          Drama, Mystery, War                 8.5         413088   
..                                     ...     ...            ...   
995     Action, Horror, Sci-Fi                 6.7         212881   
996       Action, Crime, Drama                 7.6         185112   
997   Drama, Mystery, Thriller                 7.5         328196   
998             Comedy, Horror                 7.2         137754   
999  Action, Adventure, Comedy                 7.7         547316   

    Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  Sci-Fi  \
0             $14.67M       0       0      1      1       0         0       0   
1            $353.01M       0       0      1      1       0         0       0   
2             $10.00M       0       0      1      1       0         0       0   
3              $0.05M       0       0      1      1       0         0       0   
4             $46.89M       0       0      1      0       0         0       0   
..                ...     ...     ...    ...    ...     ...       ...     ...   
995           $82.35M       1       1      0      0       0         0       1   
996           $40.90M       1       0      1      1       0         0       0   
997           $55.69M       0       0      1      0       0         1       0   
998           $32.22M       0       1      0      0       1         0       0   
999          $324.59M       1       0      0      0       1         0       0   

     Mystery  Adventure  Biography  war  
0          0          0          0    0  
1          0          0          1    0  
2          0          0          0    0  
3          1          0          0    0  
4          1          0          0    1  
..       ...        ...        ...  ...  
995        0          0          0    0  
996        0          0          0    0  
997        1          0          0    0  
998        0          0          0    0  
999        0          1          0    0  

[940 rows x 19 columns]

In [88]:
df.drop('Unnamed: 0',axis=1,inplace=True)

In [89]:
df

Title Year of Release Duration in min  \
0         1.The Godfather            1972            175    
1            2.Goodfellas            1990            145    
2          3.Pulp Fiction            1994            154    
3    4.The Usual Suspects            1995            106    
4        5.Apocalypse Now            1979            147    
..                    ...             ...             ...   
995          996.Blade II            2002            117    
996      997.Falling Down            1993            113    
997    998.Eyes Wide Shut            1999            159    
998     999.The Lost Boys            1987             97    
999      1,000.Deadpool 2            2018            119    

                                     Genre  Rating Votes in Views  \
0                 Crime, Drama                 9.2         231110   
1      Biography, Crime, Drama                 8.7         382167   
2                 Crime, Drama                 8.9          84108   
3        Crime, Drama, Mystery                 8.5          27683   
4          Drama, Mystery, War                 8.5         413088   
..                                     ...     ...            ...   
995     Action, Horror, Sci-Fi                 6.7         212881   
996       Action, Crime, Drama                 7.6         185112   
997   Drama, Mystery, Thriller                 7.5         328196   
998             Comedy, Horror                 7.2         137754   
999  Action, Adventure, Comedy                 7.7         547316   

    Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  Sci-Fi  \
0             $14.67M       0       0      1      1       0         0       0   
1            $353.01M       0       0      1      1       0         0       0   
2             $10.00M       0       0      1      1       0         0       0   
3              $0.05M       0       0      1      1       0         0       0   
4             $46.89M       0       0      1      0       0         0       0   
..                ...     ...     ...    ...    ...     ...       ...     ...   
995           $82.35M       1       1      0      0       0         0       1   
996           $40.90M       1       0      1      1       0         0       0   
997           $55.69M       0       0      1      0       0         1       0   
998           $32.22M       0       1      0      0       1         0       0   
999          $324.59M       1       0      0      0       1         0       0   

     Mystery  Adventure  Biography  war  
0          0          0          0    0  
1          0          0          1    0  
2          0          0          0    0  
3          1          0          0    0  
4          1          0          0    1  
..       ...        ...        ...  ...  
995        0          0          0    0  
996        0          0          0    0  
997        1          0          0    0  
998        0          0          0    0  
999        0          1          0    0  

[940 rows x 18 columns]

In [90]:
df['Gross in Millions'] = df['Gross in Millions'].apply(lambda x:x.replace('$',''))

In [91]:
df['Gross in Millions'] = df['Gross in Millions'].apply(lambda x:x.replace('M',''))

In [92]:
df

Title Year of Release Duration in min  \
0         1.The Godfather            1972            175    
1            2.Goodfellas            1990            145    
2          3.Pulp Fiction            1994            154    
3    4.The Usual Suspects            1995            106    
4        5.Apocalypse Now            1979            147    
..                    ...             ...             ...   
995          996.Blade II            2002            117    
996      997.Falling Down            1993            113    
997    998.Eyes Wide Shut            1999            159    
998     999.The Lost Boys            1987             97    
999      1,000.Deadpool 2            2018            119    

                                     Genre  Rating Votes in Views  \
0                 Crime, Drama                 9.2         231110   
1      Biography, Crime, Drama                 8.7         382167   
2                 Crime, Drama                 8.9          84108   
3        Crime, Drama, Mystery                 8.5          27683   
4          Drama, Mystery, War                 8.5         413088   
..                                     ...     ...            ...   
995     Action, Horror, Sci-Fi                 6.7         212881   
996       Action, Crime, Drama                 7.6         185112   
997   Drama, Mystery, Thriller                 7.5         328196   
998             Comedy, Horror                 7.2         137754   
999  Action, Adventure, Comedy                 7.7         547316   

    Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  Sci-Fi  \
0               14.67       0       0      1      1       0         0       0   
1              353.01       0       0      1      1       0         0       0   
2               10.00       0       0      1      1       0         0       0   
3                0.05       0       0      1      1       0         0       0   
4               46.89       0       0      1      0       0         0       0   
..                ...     ...     ...    ...    ...     ...       ...     ...   
995             82.35       1       1      0      0       0         0       1   
996             40.90       1       0      1      1       0         0       0   
997             55.69       0       0      1      0       0         1       0   
998             32.22       0       1      0      0       1         0       0   
999            324.59       1       0      0      0       1         0       0   

     Mystery  Adventure  Biography  war  
0          0          0          0    0  
1          0          0          1    0  
2          0          0          0    0  
3          1          0          0    0  
4          1          0          0    1  
..       ...        ...        ...  ...  
995        0          0          0    0  
996        0          0          0    0  
997        1          0          0    0  
998        0          0          0    0  
999        0          1          0    0  

[940 rows x 18 columns]

# Removing the numbers and pullstop which is present at Title

#### for that we are applying regex function

In [93]:
df['Title'][0]

'1.The Godfather'

In [94]:
import regex as re

p=re.compile(r"\.(.*)")

In [95]:
p.findall(df['Title'][0])

['The Godfather']

In [96]:
d=df['Title'].apply(lambda x:p.findall(x))

In [97]:
d=d.apply(lambda x:"".join(x))

In [98]:
df['Title']=d

In [99]:
df

Title Year of Release Duration in min  \
0         The Godfather            1972            175    
1            Goodfellas            1990            145    
2          Pulp Fiction            1994            154    
3    The Usual Suspects            1995            106    
4        Apocalypse Now            1979            147    
..                  ...             ...             ...   
995            Blade II            2002            117    
996        Falling Down            1993            113    
997      Eyes Wide Shut            1999            159    
998       The Lost Boys            1987             97    
999          Deadpool 2            2018            119    

                                     Genre  Rating Votes in Views  \
0                 Crime, Drama                 9.2         231110   
1      Biography, Crime, Drama                 8.7         382167   
2                 Crime, Drama                 8.9          84108   
3        Crime, Drama, Mystery                 8.5          27683   
4          Drama, Mystery, War                 8.5         413088   
..                                     ...     ...            ...   
995     Action, Horror, Sci-Fi                 6.7         212881   
996       Action, Crime, Drama                 7.6         185112   
997   Drama, Mystery, Thriller                 7.5         328196   
998             Comedy, Horror                 7.2         137754   
999  Action, Adventure, Comedy                 7.7         547316   

    Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  Sci-Fi  \
0               14.67       0       0      1      1       0         0       0   
1              353.01       0       0      1      1       0         0       0   
2               10.00       0       0      1      1       0         0       0   
3                0.05       0       0      1      1       0         0       0   
4               46.89       0       0      1      0       0         0       0   
..                ...     ...     ...    ...    ...     ...       ...     ...   
995             82.35       1       1      0      0       0         0       1   
996             40.90       1       0      1      1       0         0       0   
997             55.69       0       0      1      0       0         1       0   
998             32.22       0       1      0      0       1         0       0   
999            324.59       1       0      0      0       1         0       0   

     Mystery  Adventure  Biography  war  
0          0          0          0    0  
1          0          0          1    0  
2          0          0          0    0  
3          1          0          0    0  
4          1          0          0    1  
..       ...        ...        ...  ...  
995        0          0          0    0  
996        0          0          0    0  
997        1          0          0    0  
998        0          0          0    0  
999        0          1          0    0  

[940 rows x 18 columns]

# Again checking the d-types

In [100]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 940 entries, 0 to 999
Data columns (total 18 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Title              940 non-null    object 
 1   Year of Release    940 non-null    object 
 2   Duration in min    940 non-null    object 
 3   Genre              940 non-null    object 
 4   Rating             940 non-null    float64
 5   Votes in Views     940 non-null    object 
 6   Gross in Millions  940 non-null    object 
 7   Action             940 non-null    int64  
 8   Horror             940 non-null    int64  
 9   Drama              940 non-null    int64  
 10  Crime              940 non-null    int64  
 11  Comedy             940 non-null    int64  
 12  Thriller           940 non-null    int64  
 13  Sci-Fi             940 non-null    int64  
 14  Mystery            940 non-null    int64  
 15  Adventure          940 non-null    int64  
 16  Biography          940 non

#### Here we have 'Duration in min','Votes in Views','Gross in Millions' in object, so we have to convert in int.

In [101]:
df['Duration in min']=df["Duration in min"].astype(int)

In [102]:
df['Votes in Views']=df['Votes in Views'].astype(int)

In [103]:
df['Gross in Millions']=df['Gross in Millions'].astype(float)

In [104]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 940 entries, 0 to 999
Data columns (total 18 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Title              940 non-null    object 
 1   Year of Release    940 non-null    object 
 2   Duration in min    940 non-null    int32  
 3   Genre              940 non-null    object 
 4   Rating             940 non-null    float64
 5   Votes in Views     940 non-null    int32  
 6   Gross in Millions  940 non-null    float64
 7   Action             940 non-null    int64  
 8   Horror             940 non-null    int64  
 9   Drama              940 non-null    int64  
 10  Crime              940 non-null    int64  
 11  Comedy             940 non-null    int64  
 12  Thriller           940 non-null    int64  
 13  Sci-Fi             940 non-null    int64  
 14  Mystery            940 non-null    int64  
 15  Adventure          940 non-null    int64  
 16  Biography          940 non

In [105]:
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
995            Blade II            2002              117   
996        Falling Down            1993              113   
997      Eyes Wide Shut            1999              159   
998       The Lost Boys            1987               97   
999          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
995     Action, Horror, Sci-Fi                 6.7          212881   
996       Action, Crime, Drama                 7.6          185112   
997   Drama, Mystery, Thriller                 7.5          328196   
998             Comedy, Horror                 7.2          137754   
999  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
995              82.35       1       1      0      0       0         0   
996              40.90       1       0      1      1       0         0   
997              55.69       0       0      1      0       0         1   
998              32.22       0       1      0      0       1         0   
999             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
995       1        0          0          0    0  
996       0        0          0          0    0  
997       0        1          0          0    0  
998       0        0          0          0    0  
999       0        0          1          0    0  

[940 rows x 18 columns]

# we have 940 rows but in data frame we have 999 in the last row. we will reset the index.

In [106]:
df.reset_index(inplace=True)

In [107]:
df

index               Title Year of Release  Duration in min  \
0        0       The Godfather            1972              175   
1        1          Goodfellas            1990              145   
2        2        Pulp Fiction            1994              154   
3        3  The Usual Suspects            1995              106   
4        4      Apocalypse Now            1979              147   
..     ...                 ...             ...              ...   
935    995            Blade II            2002              117   
936    996        Falling Down            1993              113   
937    997      Eyes Wide Shut            1999              159   
938    998       The Lost Boys            1987               97   
939    999          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
935     Action, Horror, Sci-Fi                 6.7          212881   
936       Action, Crime, Drama                 7.6          185112   
937   Drama, Mystery, Thriller                 7.5          328196   
938             Comedy, Horror                 7.2          137754   
939  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
935              82.35       1       1      0      0       0         0   
936              40.90       1       0      1      1       0         0   
937              55.69       0       0      1      0       0         1   
938              32.22       0       1      0      0       1         0   
939             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
935       1        0          0          0    0  
936       0        0          0          0    0  
937       0        1          0          0    0  
938       0        0          0          0    0  
939       0        0          1          0    0  

[940 rows x 19 columns]

In [108]:
df.drop('index',axis=1,inplace=True)

In [109]:
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
935            Blade II            2002              117   
936        Falling Down            1993              113   
937      Eyes Wide Shut            1999              159   
938       The Lost Boys            1987               97   
939          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
935     Action, Horror, Sci-Fi                 6.7          212881   
936       Action, Crime, Drama                 7.6          185112   
937   Drama, Mystery, Thriller                 7.5          328196   
938             Comedy, Horror                 7.2          137754   
939  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
935              82.35       1       1      0      0       0         0   
936              40.90       1       0      1      1       0         0   
937              55.69       0       0      1      0       0         1   
938              32.22       0       1      0      0       1         0   
939             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
935       1        0          0          0    0  
936       0        0          0          0    0  
937       0        1          0          0    0  
938       0        0          0          0    0  
939       0        0          1          0    0  

[940 rows x 18 columns]

# Finally checking the Nan Values are present or not

In [110]:
df.isna().sum()

Title                0
Year of Release      0
Duration in min      0
Genre                0
Rating               0
Votes in Views       0
Gross in Millions    0
Action               0
Horror               0
Drama                0
Crime                0
Comedy               0
Thriller             0
Sci-Fi               0
Mystery              0
Adventure            0
Biography            0
war                  0
dtype: int64

In [111]:
df['Year of Release']

0      1972
1      1990
2      1994
3      1995
4      1979
       ... 
935    2002
936    1993
937    1999
938    1987
939    2018
Name: Year of Release, Length: 940, dtype: object

# so there are some string values in year of release,we have to clean that too

In [112]:
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
935            Blade II            2002              117   
936        Falling Down            1993              113   
937      Eyes Wide Shut            1999              159   
938       The Lost Boys            1987               97   
939          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
935     Action, Horror, Sci-Fi                 6.7          212881   
936       Action, Crime, Drama                 7.6          185112   
937   Drama, Mystery, Thriller                 7.5          328196   
938             Comedy, Horror                 7.2          137754   
939  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
935              82.35       1       1      0      0       0         0   
936              40.90       1       0      1      1       0         0   
937              55.69       0       0      1      0       0         1   
938              32.22       0       1      0      0       1         0   
939             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
935       1        0          0          0    0  
936       0        0          0          0    0  
937       0        1          0          0    0  
938       0        0          0          0    0  
939       0        0          1          0    0  

[940 rows x 18 columns]

In [113]:
df['Year of Release']= df['Year of Release']
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
935            Blade II            2002              117   
936        Falling Down            1993              113   
937      Eyes Wide Shut            1999              159   
938       The Lost Boys            1987               97   
939          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
935     Action, Horror, Sci-Fi                 6.7          212881   
936       Action, Crime, Drama                 7.6          185112   
937   Drama, Mystery, Thriller                 7.5          328196   
938             Comedy, Horror                 7.2          137754   
939  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
935              82.35       1       1      0      0       0         0   
936              40.90       1       0      1      1       0         0   
937              55.69       0       0      1      0       0         1   
938              32.22       0       1      0      0       1         0   
939             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
935       1        0          0          0    0  
936       0        0          0          0    0  
937       0        1          0          0    0  
938       0        0          0          0    0  
939       0        0          1          0    0  

[940 rows x 18 columns]

In [114]:
df = df[df['Year of Release']!='I) ']
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
935            Blade II            2002              117   
936        Falling Down            1993              113   
937      Eyes Wide Shut            1999              159   
938       The Lost Boys            1987               97   
939          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
935     Action, Horror, Sci-Fi                 6.7          212881   
936       Action, Crime, Drama                 7.6          185112   
937   Drama, Mystery, Thriller                 7.5          328196   
938             Comedy, Horror                 7.2          137754   
939  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
935              82.35       1       1      0      0       0         0   
936              40.90       1       0      1      1       0         0   
937              55.69       0       0      1      0       0         1   
938              32.22       0       1      0      0       1         0   
939             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
935       1        0          0          0    0  
936       0        0          0          0    0  
937       0        1          0          0    0  
938       0        0          0          0    0  
939       0        0          1          0    0  

[915 rows x 18 columns]

In [115]:
df['Year of Release']= df['Year of Release']
df

C:\Users\DEVEND~1\AppData\Local\Temp/ipykernel_28060/1509194407.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Year of Release']= df['Year of Release']


Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
935            Blade II            2002              117   
936        Falling Down            1993              113   
937      Eyes Wide Shut            1999              159   
938       The Lost Boys            1987               97   
939          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
935     Action, Horror, Sci-Fi                 6.7          212881   
936       Action, Crime, Drama                 7.6          185112   
937   Drama, Mystery, Thriller                 7.5          328196   
938             Comedy, Horror                 7.2          137754   
939  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
935              82.35       1       1      0      0       0         0   
936              40.90       1       0      1      1       0         0   
937              55.69       0       0      1      0       0         1   
938              32.22       0       1      0      0       1         0   
939             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
935       1        0          0          0    0  
936       0        0          0          0    0  
937       0        1          0          0    0  
938       0        0          0          0    0  
939       0        0          1          0    0  

[915 rows x 18 columns]

In [116]:
df = df[df['Year of Release']!='500) Days of Summer\n']

In [117]:
df['Year of Release']= df['Year of Release']

C:\Users\DEVEND~1\AppData\Local\Temp/ipykernel_28060/943903745.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Year of Release']= df['Year of Release']


In [134]:
df = df[df['Year of Release']!='1985–1998\nEpisode:\nThe Firm\n']
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
901            Blade II            2002              117   
902        Falling Down            1993              113   
903      Eyes Wide Shut            1999              159   
904       The Lost Boys            1987               97   
905          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
901     Action, Horror, Sci-Fi                 6.7          212881   
902       Action, Crime, Drama                 7.6          185112   
903   Drama, Mystery, Thriller                 7.5          328196   
904             Comedy, Horror                 7.2          137754   
905  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
901              82.35       1       1      0      0       0         0   
902              40.90       1       0      1      1       0         0   
903              55.69       0       0      1      0       0         1   
904              32.22       0       1      0      0       1         0   
905             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
901       1        0          0          0    0  
902       0        0          0          0    0  
903       0        1          0          0    0  
904       0        0          0          0    0  
905       0        0          1          0    0  

[906 rows x 18 columns]

In [135]:
df['Year of Release']= df['Year of Release']
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
901            Blade II            2002              117   
902        Falling Down            1993              113   
903      Eyes Wide Shut            1999              159   
904       The Lost Boys            1987               97   
905          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
901     Action, Horror, Sci-Fi                 6.7          212881   
902       Action, Crime, Drama                 7.6          185112   
903   Drama, Mystery, Thriller                 7.5          328196   
904             Comedy, Horror                 7.2          137754   
905  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
901              82.35       1       1      0      0       0         0   
902              40.90       1       0      1      1       0         0   
903              55.69       0       0      1      0       0         1   
904              32.22       0       1      0      0       1         0   
905             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
901       1        0          0          0    0  
902       0        0          0          0    0  
903       0        1          0          0    0  
904       0        0          0          0    0  
905       0        0          1          0    0  

[906 rows x 18 columns]

In [137]:
df = df[df['Year of Release']!='II) ']
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
901            Blade II            2002              117   
902        Falling Down            1993              113   
903      Eyes Wide Shut            1999              159   
904       The Lost Boys            1987               97   
905          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
901     Action, Horror, Sci-Fi                 6.7          212881   
902       Action, Crime, Drama                 7.6          185112   
903   Drama, Mystery, Thriller                 7.5          328196   
904             Comedy, Horror                 7.2          137754   
905  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
901              82.35       1       1      0      0       0         0   
902              40.90       1       0      1      1       0         0   
903              55.69       0       0      1      0       0         1   
904              32.22       0       1      0      0       1         0   
905             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
901       1        0          0          0    0  
902       0        0          0          0    0  
903       0        1          0          0    0  
904       0        0          0          0    0  
905       0        0          1          0    0  

[906 rows x 18 columns]

In [138]:
df['Year of Release']= df['Year of Release']
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
901            Blade II            2002              117   
902        Falling Down            1993              113   
903      Eyes Wide Shut            1999              159   
904       The Lost Boys            1987               97   
905          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
901     Action, Horror, Sci-Fi                 6.7          212881   
902       Action, Crime, Drama                 7.6          185112   
903   Drama, Mystery, Thriller                 7.5          328196   
904             Comedy, Horror                 7.2          137754   
905  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
901              82.35       1       1      0      0       0         0   
902              40.90       1       0      1      1       0         0   
903              55.69       0       0      1      0       0         1   
904              32.22       0       1      0      0       1         0   
905             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
901       1        0          0          0    0  
902       0        0          0          0    0  
903       0        1          0          0    0  
904       0        0          0          0    0  
905       0        0          1          0    0  

[906 rows x 18 columns]

In [139]:
df = df[df['Year of Release']!='The Unexpected Virtue of Ignorance']
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
901            Blade II            2002              117   
902        Falling Down            1993              113   
903      Eyes Wide Shut            1999              159   
904       The Lost Boys            1987               97   
905          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
901     Action, Horror, Sci-Fi                 6.7          212881   
902       Action, Crime, Drama                 7.6          185112   
903   Drama, Mystery, Thriller                 7.5          328196   
904             Comedy, Horror                 7.2          137754   
905  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
901              82.35       1       1      0      0       0         0   
902              40.90       1       0      1      1       0         0   
903              55.69       0       0      1      0       0         1   
904              32.22       0       1      0      0       1         0   
905             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
901       1        0          0          0    0  
902       0        0          0          0    0  
903       0        1          0          0    0  
904       0        0          0          0    0  
905       0        0          1          0    0  

[906 rows x 18 columns]

In [140]:
df['Year of Release']= df['Year of Release']
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
901            Blade II            2002              117   
902        Falling Down            1993              113   
903      Eyes Wide Shut            1999              159   
904       The Lost Boys            1987               97   
905          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
901     Action, Horror, Sci-Fi                 6.7          212881   
902       Action, Crime, Drama                 7.6          185112   
903   Drama, Mystery, Thriller                 7.5          328196   
904             Comedy, Horror                 7.2          137754   
905  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
901              82.35       1       1      0      0       0         0   
902              40.90       1       0      1      1       0         0   
903              55.69       0       0      1      0       0         1   
904              32.22       0       1      0      0       1         0   
905             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
901       1        0          0          0    0  
902       0        0          0          0    0  
903       0        1          0          0    0  
904       0        0          0          0    0  
905       0        0          1          0    0  

[906 rows x 18 columns]

In [141]:
df = df[df['Year of Release']!='III) ']
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
901            Blade II            2002              117   
902        Falling Down            1993              113   
903      Eyes Wide Shut            1999              159   
904       The Lost Boys            1987               97   
905          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
901     Action, Horror, Sci-Fi                 6.7          212881   
902       Action, Crime, Drama                 7.6          185112   
903   Drama, Mystery, Thriller                 7.5          328196   
904             Comedy, Horror                 7.2          137754   
905  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
901              82.35       1       1      0      0       0         0   
902              40.90       1       0      1      1       0         0   
903              55.69       0       0      1      0       0         1   
904              32.22       0       1      0      0       1         0   
905             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
901       1        0          0          0    0  
902       0        0          0          0    0  
903       0        1          0          0    0  
904       0        0          0          0    0  
905       0        0          1          0    0  

[906 rows x 18 columns]

In [142]:
df['Year of Release']= df['Year of Release']
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
901            Blade II            2002              117   
902        Falling Down            1993              113   
903      Eyes Wide Shut            1999              159   
904       The Lost Boys            1987               97   
905          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
901     Action, Horror, Sci-Fi                 6.7          212881   
902       Action, Crime, Drama                 7.6          185112   
903   Drama, Mystery, Thriller                 7.5          328196   
904             Comedy, Horror                 7.2          137754   
905  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
901              82.35       1       1      0      0       0         0   
902              40.90       1       0      1      1       0         0   
903              55.69       0       0      1      0       0         1   
904              32.22       0       1      0      0       1         0   
905             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
901       1        0          0          0    0  
902       0        0          0          0    0  
903       0        1          0          0    0  
904       0        0          0          0    0  
905       0        0          1          0    0  

[906 rows x 18 columns]

In [126]:
df.info()

# finally we converted the year of release column into int d-type

<class 'pandas.core.frame.DataFrame'>
Int64Index: 906 entries, 0 to 939
Data columns (total 18 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Title              906 non-null    object 
 1   Year of Release    906 non-null    object 
 2   Duration in min    906 non-null    int32  
 3   Genre              906 non-null    object 
 4   Rating             906 non-null    float64
 5   Votes in Views     906 non-null    int32  
 6   Gross in Millions  906 non-null    float64
 7   Action             906 non-null    int64  
 8   Horror             906 non-null    int64  
 9   Drama              906 non-null    int64  
 10  Crime              906 non-null    int64  
 11  Comedy             906 non-null    int64  
 12  Thriller           906 non-null    int64  
 13  Sci-Fi             906 non-null    int64  
 14  Mystery            906 non-null    int64  
 15  Adventure          906 non-null    int64  
 16  Biography          906 non

In [127]:
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
935            Blade II            2002              117   
936        Falling Down            1993              113   
937      Eyes Wide Shut            1999              159   
938       The Lost Boys            1987               97   
939          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
935     Action, Horror, Sci-Fi                 6.7          212881   
936       Action, Crime, Drama                 7.6          185112   
937   Drama, Mystery, Thriller                 7.5          328196   
938             Comedy, Horror                 7.2          137754   
939  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
935              82.35       1       1      0      0       0         0   
936              40.90       1       0      1      1       0         0   
937              55.69       0       0      1      0       0         1   
938              32.22       0       1      0      0       1         0   
939             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
935       1        0          0          0    0  
936       0        0          0          0    0  
937       0        1          0          0    0  
938       0        0          0          0    0  
939       0        0          1          0    0  

[906 rows x 18 columns]

In [128]:
df.reset_index(inplace=True)

In [129]:
df

index               Title Year of Release  Duration in min  \
0        0       The Godfather            1972              175   
1        1          Goodfellas            1990              145   
2        2        Pulp Fiction            1994              154   
3        3  The Usual Suspects            1995              106   
4        4      Apocalypse Now            1979              147   
..     ...                 ...             ...              ...   
901    935            Blade II            2002              117   
902    936        Falling Down            1993              113   
903    937      Eyes Wide Shut            1999              159   
904    938       The Lost Boys            1987               97   
905    939          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
901     Action, Horror, Sci-Fi                 6.7          212881   
902       Action, Crime, Drama                 7.6          185112   
903   Drama, Mystery, Thriller                 7.5          328196   
904             Comedy, Horror                 7.2          137754   
905  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
901              82.35       1       1      0      0       0         0   
902              40.90       1       0      1      1       0         0   
903              55.69       0       0      1      0       0         1   
904              32.22       0       1      0      0       1         0   
905             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
901       1        0          0          0    0  
902       0        0          0          0    0  
903       0        1          0          0    0  
904       0        0          0          0    0  
905       0        0          1          0    0  

[906 rows x 19 columns]

In [130]:
df.drop('index',axis=1,inplace=True)

In [131]:
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
901            Blade II            2002              117   
902        Falling Down            1993              113   
903      Eyes Wide Shut            1999              159   
904       The Lost Boys            1987               97   
905          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
901     Action, Horror, Sci-Fi                 6.7          212881   
902       Action, Crime, Drama                 7.6          185112   
903   Drama, Mystery, Thriller                 7.5          328196   
904             Comedy, Horror                 7.2          137754   
905  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
901              82.35       1       1      0      0       0         0   
902              40.90       1       0      1      1       0         0   
903              55.69       0       0      1      0       0         1   
904              32.22       0       1      0      0       1         0   
905             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
901       1        0          0          0    0  
902       0        0          0          0    0  
903       0        1          0          0    0  
904       0        0          0          0    0  
905       0        0          1          0    0  

[906 rows x 18 columns]

In [132]:
df.to_csv(r"C:\Users\devender bahadur\Downloads\Web Scrabing Final Project\cleaned_df.csv")

In [133]:
df

Title Year of Release  Duration in min  \
0         The Godfather            1972              175   
1            Goodfellas            1990              145   
2          Pulp Fiction            1994              154   
3    The Usual Suspects            1995              106   
4        Apocalypse Now            1979              147   
..                  ...             ...              ...   
901            Blade II            2002              117   
902        Falling Down            1993              113   
903      Eyes Wide Shut            1999              159   
904       The Lost Boys            1987               97   
905          Deadpool 2            2018              119   

                                     Genre  Rating  Votes in Views  \
0                 Crime, Drama                 9.2          231110   
1      Biography, Crime, Drama                 8.7          382167   
2                 Crime, Drama                 8.9           84108   
3        Crime, Drama, Mystery                 8.5           27683   
4          Drama, Mystery, War                 8.5          413088   
..                                     ...     ...             ...   
901     Action, Horror, Sci-Fi                 6.7          212881   
902       Action, Crime, Drama                 7.6          185112   
903   Drama, Mystery, Thriller                 7.5          328196   
904             Comedy, Horror                 7.2          137754   
905  Action, Adventure, Comedy                 7.7          547316   

     Gross in Millions  Action  Horror  Drama  Crime  Comedy  Thriller  \
0                14.67       0       0      1      1       0         0   
1               353.01       0       0      1      1       0         0   
2                10.00       0       0      1      1       0         0   
3                 0.05       0       0      1      1       0         0   
4                46.89       0       0      1      0       0         0   
..                 ...     ...     ...    ...    ...     ...       ...   
901              82.35       1       1      0      0       0         0   
902              40.90       1       0      1      1       0         0   
903              55.69       0       0      1      0       0         1   
904              32.22       0       1      0      0       1         0   
905             324.59       1       0      0      0       1         0   

     Sci-Fi  Mystery  Adventure  Biography  war  
0         0        0          0          0    0  
1         0        0          0          1    0  
2         0        0          0          0    0  
3         0        1          0          0    0  
4         0        1          0          0    1  
..      ...      ...        ...        ...  ...  
901       1        0          0          0    0  
902       0        0          0          0    0  
903       0        1          0          0    0  
904       0        0          0          0    0  
905       0        0          1          0    0  

[906 rows x 18 columns]